examples/chip-tool/commands/clusters/ModelCommand.cpp

```diff
@@ -37,39 +37,28 @@ void DispatchSingleClusterCommand(chip::ClusterId aClusterId, chip::CommandId aC
         "Default DispatchSingleClusterCommand is called, this should be replaced by actual dispatched for cluster commands");
 }
 
-CHIP_ERROR ModelCommand::Run(PersistentStorage & storage, NodeId localId, NodeId remoteId)
+CHIP_ERROR ModelCommand::Run(NodeId localId, NodeId remoteId)
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
 
-    chip::Controller::CommissionerInitParams initParams;
-    initParams.storageDelegate = &storage;
-
-    err = mOpCredsIssuer.Initialize(storage);
-    VerifyOrExit(err == CHIP_NO_ERROR, ChipLogError(Controller, "Init failure! Operational Cred Issuer: %s", ErrorStr(err)));
-
-    initParams.operationalCredentialsDelegate = &mOpCredsIssuer;
-
-    err = mCommissioner.SetUdpListenPort(storage.GetListenPort());
-    VerifyOrExit(err == CHIP_NO_ERROR, ChipLogError(Controller, "Init failure! Commissioner: %s", ErrorStr(err)));
-
-    err = mCommissioner.Init(localId, initParams);
-    VerifyOrExit(err == CHIP_NO_ERROR, ChipLogError(Controller, "Init failure! Commissioner: %s", ErrorStr(err)));
-
-    err = mCommissioner.ServiceEvents();
-    VerifyOrExit(err == CHIP_NO_ERROR, ChipLogError(Controller, "Init failure! Run Loop: %s", ErrorStr(err)));
-
-    err = mCommissioner.GetDevice(remoteId, &mDevice);
+    chip::DeviceLayer::PlatformMgr().LockChipStack();
+    err = GetExecContext()->Commissioner->GetDevice(remoteId, &mDevice);
+    chip::DeviceLayer::PlatformMgr().UnlockChipStack();
+    
     VerifyOrExit(err == CHIP_NO_ERROR, ChipLogError(chipTool, "Init failure! No pairing for device: %" PRIu64, localId));
 
     UpdateWaitForResponse(true);
+
+    chip::DeviceLayer::PlatformMgr().LockChipStack();
     err = SendCommand(mDevice, mEndPointId);
+    chip::DeviceLayer::PlatformMgr().UnlockChipStack();
```

Comment:
Q: Is there an RAII'fied version of this lock I could use?

Reply:
We should build one. Next PR?

Reply:
Here you go. Put this just below PlatformMgr in PlatformManager.h

```
/**
 * RAII locking for PlatformManager to simplify management of
 * LockChipStack()/UnlockChipStack calls.
 */
class StackLocker
{
public:
    StackLocker(PlatformManager & aPlatformManager) : mPlatformManager(aPlatformManager) 
    {
        mPlatformManager.LockChipStack();
    }
    
    ~StackLocker() 
    { 
        mPlatformManager.UnlockChipStack();
    }

private:
    PlatformManager & mPlatformManager;
};
```